core/src/test/java/org/elasticsearch/script/StoredScriptTests.java

```diff
@@ -0,0 +1,333 @@
+        assertThat(smd.getStoredScript("test0", "lang2"), equalTo(source2));
+        assertThat(smd.getStoredScript("test3", null), equalTo(source3));
+        assertThat(smd.getStoredScript("test3", "lang1"), equalTo(source3));
+
+        smd = ScriptMetaData.deleteStoredScript(smd, "test0", "lang1");
+
+        assertThat(smd.getStoredScript("test0", null), equalTo(source2));
+        assertThat(smd.getStoredScript("test0", "lang0"), equalTo(source0));
+        assertThat(smd.getStoredScript("test0", "lang1"), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang2"), equalTo(source2));
+        assertThat(smd.getStoredScript("test3", null), equalTo(source3));
+        assertThat(smd.getStoredScript("test3", "lang1"), equalTo(source3));
+
+        smd = ScriptMetaData.deleteStoredScript(smd, "test0", null);
+
+        assertThat(smd.getStoredScript("test0", null), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang0"), equalTo(source0));
+        assertThat(smd.getStoredScript("test0", "lang1"), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang2"), nullValue());
+        assertThat(smd.getStoredScript("test3", null), equalTo(source3));
+        assertThat(smd.getStoredScript("test3", "lang1"), equalTo(source3));
+
+        smd = ScriptMetaData.deleteStoredScript(smd, "test3", "lang1");
+
+        assertThat(smd.getStoredScript("test0", null), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang0"), equalTo(source0));
+        assertThat(smd.getStoredScript("test0", "lang1"), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang2"), nullValue());
+        assertThat(smd.getStoredScript("test3", null), nullValue());
+        assertThat(smd.getStoredScript("test3", "lang1"), nullValue());
+
+        smd = ScriptMetaData.deleteStoredScript(smd, "test0", "lang0");
+
+        assertThat(smd.getStoredScript("test0", null), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang0"), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang1"), nullValue());
+        assertThat(smd.getStoredScript("test0", "lang2"), nullValue());
+        assertThat(smd.getStoredScript("test3", null), nullValue());
+        assertThat(smd.getStoredScript("test3", "lang1"), nullValue());
+    }
+
+    public void testInvalidDelete() {
+        ResourceNotFoundException rnfe =
+            expectThrows(ResourceNotFoundException.class, () -> ScriptMetaData.deleteStoredScript(null, "test", "lang"));
+        assertThat(rnfe.getMessage(), equalTo("stored script [test] using lang [lang] does not exist and cannot be deleted"));
+
+        rnfe = expectThrows(ResourceNotFoundException.class, () -> ScriptMetaData.deleteStoredScript(null, "test", null));
+        assertThat(rnfe.getMessage(), equalTo("stored script [test] does not exist and cannot be deleted"));
+    }
+
+    public void testSourceParsing() throws Exception {
```

Comment:
@colings86 is adding base test classes for writeable and xcontent serialization / de-serialization via #22281

That should get merged quickly. When that is in can you make this test class extend from `AbstractSerializingTestCase`. That will add random serialization tests in addition that the serialization  that exist already.